docs/DeveloperGuide.md

```diff
@@ -256,7 +269,103 @@ as values.<br>
+
+ As seen in the sequence diagram above, which shows the process of creating a `BirthDate` object.
+ `ParserUtil#parseBirthDate(validDate)` is called, with the user supplying a `validDate` in the form of a String.
+The method will process the `validDate` into a `String` called `trimmedDate` and call 
+ `DocumentedDate#isValidDate(trimmedDate)` to check if it is a valid date. if the `trimmedDate` is indeed a valid date,
+it will then be used to create a new `BirthDate` object. After that, a final check is done using the `getDaysPassed()` 
+ method, before `ParserUtil` returns the newly created `BirthDate` object. For `BirthDate` objects, the check using the 
+ `getDaysPassed()` method has to ensure that `BirthDate` objects are not created using dates in the future, 
+ i.e. either past dates or the current date.
+
+As the `RecentDate` and `ReminderDate` objects have similar requirements to the `BirthDate` objects, the process of 
+creating these objects are the same as the `BirthDate` objects. For better comparison, let us examine one more sequence
+diagram for the creation of a `ReminderDate` object.
+
+<img src="images/ReminderDateCreationSequenceDiagram.png" width="450" />
+
+As seen above, the process exactly mirrors that of the `BirthDate` object. However, the biggest difference is that the
+check using `getDaysPassed()` must make sure that `ReminderDate` objects cannot be created with a date that has passed.
+This is because the function of the reminder is to ensure that users can keep track of tasks that have not yet been done
+or important events that are upcoming.
+
+The `RecentDate` objects are much more similar to the `BirthDate` objects as they have the same requirement of not being
+able to be created using a date in the future. The rationale is because `RecentDate` keep track of the user's 
+interaction with clients and contacts, and the user must have interacted with the contact before they save the 
+interaction record in the application. 
+
+Having two levels of checks ensure that when the working "date" type objects are less bug prone due when using the specific
+"date" type object for their designated usages. This makes features such as `after` or `within` return valid entries
+when used, instead of an invalid or unexpected entry.
+
+#### Design Consideration
+
+##### Aspect: How date type objects are created
+
+* Alternative 1 (current implementation): "Date" type objects extend `DocumentedDate`.<br>
+    * Pros: We can use polymorphism to modify certain behaviour of each specific child class (i.e. Allowing `BirthDate` 
+      objects to be read as recurring dates when necessary.)
+    * Less bug prone as each child class is independent of each other.
+    * Cons: More overhead as more classes are required, possibly more coupling if code is not written with 
+      Liskov Substitution Principle in mind.
+   
+ 
+* Alternative 2: Use `DocumentedDate`, but use other components to differentiate (i.e. `enum` types)
```

Comment:
elaborate on this? I don't really understand. Use `DocumentedDate` as the actual date?